great_expectations/self_check/util.py

```diff
@@ -1904,10 +1904,10 @@ def check_json_test_result(test, result, data_asset=None):
                 # check if value can be sorted; if so, sort so arbitrary ordering of results does not cause failure
                 if (isinstance(value, list)) & (len(value) >= 1):
                     if type(value[0].__lt__(value[0])) != type(NotImplemented):
-                        value = value.sort()
-                        result["result"]["unexpected_list"] = result["result"][
-                            "unexpected_list"
-                        ].sort()
+                        value = sorted(value, key=lambda x: str(x))
+                        result["result"]["unexpected_list"] = sorted(
+                            result["result"]["unexpected_list"], key=lambda x: str(x)
+                        )
```

Comment:
The types for both of these should be `list`. We do a check on L1905 to see if `value` is a list, and we only get here if that resolves to True. We also assume that `unexpected_list` is True, and I think that's just a fundamental part of expectations (I'm not sure whether we test than anywhere). I can add some typing here as well - I'm not going to add it everywhere in this module, so it may look a bit funny, but if adding it here makes it clearer, I can do that.

Reply:
Nope I think that explanation is more than sufficient for me :) thanks for clarifying.

Reply:
Ah, and the custom comparator is due to an issue that was arising if the values in the list were of different types. So if one of the values was a list, but the rest were strings, this would raise an error due to trying to sort items of different types.